server/i18n/en.json

```diff
@@ -8674,7 +8674,7 @@
   },
   {
     "id": "model.channel.is_valid.display_name.app_error",
-    "translation": "Invalid display name."
+    "translation": "Invalid display name. Must be {{.MaxRunes}} or fewer characters (current: {{.ActualRunes}})."
```

Comment:
Changing the translation string also means that all translations need to get updated, which is a lot of work for the community. It's also not AI-friendly (something that might be keen to this effort) to require updates to `i18n/en.json`.

I wonder if we should change the signature of the `IsValid()` methods to `error`, where it is possible.

Reply:
I get that it's a lot of work. But I also don't see any other way. If we need to improve the log messages, this is a cost that we need to pay. Totally open to any ideas you have.

> I wonder if we should change the signature of the IsValid() methods to error, where it is possible.

Yes, absolutely. I mentioned this in my confluence page.